cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -216,6 +216,33 @@ void CheckRoundTripResult(const std::shared_ptr<Table> expected_table,
   compute::AssertTablesEqualIgnoringOrder(merged_expected, output_table);
 }
 
+int CountProjectNodeOptionsInDeclarations(const compute::Declaration& input) {
+  int counter = 0;
+  if (input.factory_name == "project") {
+    counter++;
+  }
+  const auto& inputs = input.inputs;
+  for (const auto& in : inputs) {
+    return counter +
+           CountProjectNodeOptionsInDeclarations(std::get<compute::Declaration>(in));
+  }
+  return counter;
+}
```

Comment:
```suggestion
int CountProjectNodeOptionsInDeclarations(const compute::Declaration& input) {
  int counter = 0;
  if (input.factory_name == "project") {
    counter++;
  }
  const auto& inputs = input.inputs;
  for (const auto& in : inputs) {
    counter += CountProjectNodeOptionsInDeclarations(std::get<compute::Declaration>(in));
  }
  return counter;
}
```

Reply:
Did the previous code only work because the max number of projections was 2?

Reply:
Btw, I updated the PR to include this suggestion.

Reply:
For it to fail you would need a node with multiple inputs.  So something like a join where there was a project on the right hand side.